### ISSUE TITLE
[Support]

### ISSUE THREAD

**Issue by gennaro995** [DirectoryTree/LdapRecord#367]
**Environment:**
 - LDAP Server Type: [ActiveDirectory]
 - PHP Version: [7.4.22]
 - LdapRecord Version: v2.5.0

Hi, first of all thanks for this wonderful package!
But i'm having some troubles with filther in search phase after the binding:
Let me explain better, i set-up all the files as Follow:

*in auth.php

 'users' => [
            'driver' => 'ldap',
            'model' => LdapRecord\Models\OpenLDAP\User::class,
            'rules' => [
                App\Ldap\Rules\OnlyDefinedUsers::class,
            ],
            'database' => [
                'model' => App\Models\User::class,
                'sync_passwords' => false,
                'sync_attributes' => [
                    'name' => 'givenName',
                    'username' => 'uid',
                    'email' => 'givenName'
                ],
                'sync_existing' => [
                    'username' => 'uid',
                ],
                'password_column' => 'password',
            ],
        ],

defined User:class*

    protected $passwordAttribute = 'userpassword';
    protected $passwordHashMethod = 'ssha';

    public static $objectClasses = [
        'top',
        'person',
        'organizationalperson',
        'inetorgperson',
        'ericssonuserauthorization',
    ];
    protected $dates = [
        'lastlogon' => 'windows-int',
        'lastlogoff' => 'windows-int',
        'pwdlastset' => 'windows-int',
        'lockouttime' => 'windows-int',
        'accountexpires' => 'windows-int',
        'badpasswordtime' => 'windows-int',
        'lastlogontimestamp' => 'windows-int',
        'ds-pwp-last-login-time' => 'windows-int',
    ];

    protected static function boot()
    {
        parent::boot();
        static::addGlobalScope(new RejectComputerObjectClass());
    }

    public function groups()
    {
        return $this->hasMany(Group::class, 'member')->with($this->primaryGroup());
    }

    public function manager()
    {
        return $this->hasOne(static::class, 'manager');
    }

    public function primaryGroup()
    {
        return $this->hasOnePrimaryGroup(Group::class, 'primarygroupid');
    }

    public function scopeWhereHasMailbox(Builder $query)
    {
        return $query->whereHas('msExchMailboxGuid');
    }
}

but only bind phase go well, and stops on search so seems that the Filter doesn't match.
Instead if i search from terminal calling from php artisan tinker all seems working fine.

this is results from laravel.log: (i hide some sensitive info)

[2021-12-06 17:40:05] local.INFO: LDAP (ldaps://*hidden*:636) - Operation: Binding - Username: uid=*hidden*,ou=administrators,ou=topologymanagement,o=netscaperoot  
[2021-12-06 17:40:06] local.INFO: LDAP (ldaps://*hidden*:636) - Operation: Bound - Username: uid=*hidden*,ou=administrators,ou=topologymanagement,o=netscaperoot  
[2021-12-06 17:40:06] local.INFO: LDAP (ldaps://*hidden*:636) - Operation: Search - Base DN:  - Filter: (&(objectclass=\74\6f\70)(objectclass=\70\65\72\73\6f\6e)(objectclass=\6f\72\67\61\6e\69\7a\61\74\69\6f\6e\61\6c\70\65\72\73\6f\6e)(objectclass=\69\6e\65\74\6f\72\67\70\65\72\73\6f\6e)(uid=\69\63\6d\74\2d\74\65\73\74)) - Selected: (entryuuid,*) - Time Elapsed: 368.05  

Any suggestion on this?

Thanks
Gennaro

**Comment by stevebauman** [comment]
Hi there @gennaro995, could you please format your code examples and logs so they are easier to read and then I can help you tackle this issue?

**Comment by stevebauman** [comment]
No worries @gennaro995.  Are you sure you're connecting to an ActiveDirectory server? `o=netscaperoot` inside of the DN you're binding with would have me think otherwise which is why the filter would be returning no results. Can you confirm that with me?

**Comment by gennaro995** [comment]
Sorry, I edited the post, but we are using an openldap server witch add some new extra objectClass to default LDAP objectClasses.

```
/**
     * The object classes of the LDAP model.
     *
     * @var array
     */
    public static $objectClasses = [
        'top',
        'person',
        'organizationalperson',
        'inetorgperson',
        'customobject',
    ];
```

and ours base_dn is config as follow below in ldap.php:

`   'base_dn' => env('dc=icmt,dc=infra,dc=itaca,dc=local'),`

**Comment by stevebauman** [comment]
Did you insert all of this code into your Eloquent model? If so, this won't work. These models are completely independent and don't relate to each other. If you need to adjust the built-in LdapRecord models, you must extend them and then update your `auth.php` config:

```php
// app/Ldap/Models/User.php

use LdapRecord\Models\OpenLDAP\User as BaseUser;

class User extends BaseUser
{
    // ...
}
```

```php
'providers' => [
    // ...

    'users' => [
        // ...
        'model' => App\Ldap\Models\User::class,
        'database' => [
            // ...
        ],
    ],
],
```

**Comment by stevebauman** [comment]
Also can you explain this further:

> But i'm having some troubles with filther in search phase after the binding:

What exactly is incorrect about the filter? Have you tested this filter on your LDAP server manually to ensure it returns the proper results?

**Comment by gennaro995** [comment]
> Did you insert all of this code into your Eloquent model? If so, this won't work. These models are completely independent and don't relate to each other. If you need to adjust the built-in LdapRecord models, you must extend them and then update your `auth.php` config:
> 
> ```
> // app/Ldap/Models/User.php
> 
> use LdapRecord\Models\OpenLDAP\User as BaseUser;
> 
> class User extends BaseUser
> {
>     // ...
> }
> ```
> 
> ```
> 'providers' => [
>     // ...
> 
>     'users' => [
>         // ...
>         'model' => App\Ldap\Models\User::class,
>         'database' => [
>             // ...
>         ],
>     ],
> ],
> ```

I tried to fix as you suggested but seems not working.

**Comment by gennaro995** [comment]
> Also can you explain this further:
> 
> > But i'm having some troubles with filther in search phase after the binding:
> 
> What exactly is incorrect about the filter? Have you tested this filter on your LDAP server manually to ensure it returns the proper results?

Yes, if i try to search user using same filter but from bash, using command php artisan ldap browse it works.
I want also to tell that i'm using LdapRecord with Laravel Jetstream.

**Comment by gennaro995** [comment]
@stevebauman  any idea?

**Comment by stevebauman** [comment]
Closing as a stale issue -- if you're still encountering this, let me know and I can re-open.
